analyses/snv-callers/util/plot_functions.R

```diff
@@ -0,0 +1,274 @@
+  type_count_df <- maf_annot %>%
+    dplyr::group_by(type, experimental_strategy) %>%
+    dplyr::summarise(type_count = dplyr::n()) %>%
+    dplyr::filter(type_count > filter_cutoff)
+
+  # Plot this as a barplot
+  barplot <- ggplot2::ggplot(
+    type_count_df,
+    ggplot2::aes(x = reorder(type, -type_count), y = type_count)
+  )
+
+  # Get rid of legend if both data aren't being plotted
+  if (exp_strategy != "BOTH") {
+    barplot <- barplot +
+      ggplot2::geom_bar(
+        position = "dodge", stat = "identity",
+        fill = "navyblue"
+      ) +
+      ggplot2::theme(legend.position = "none")
+  } else {
+    barplot <- barplot +
+      ggplot2::geom_bar(
+        position = "dodge", stat = "identity",
+        ggplot2::aes(fill = experimental_strategy)
+      )
+  }
+
+  # Add some other aesthetics
+  barplot <- barplot +
+    ggplot2::theme_classic() +
+    ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
+    ggplot2::xlab("") +
+    ggplot2::ylab("Count") +
+    colorblindr::scale_fill_OkabeIto()
+
+  return(barplot)
+}
+
+cosmic_plot <- function(vaf_df, exp_strategy = "BOTH", cosmic_clean_file = NULL) {
+  # Plot the VAF for COSMIC vs non-COSMIC mutations
+  #
+  # Args:
+  #   vaf_df: Need a MAF data.frame that has `vaf`, `base_change` and
+  #           `experimental_strategy` columns that are added with the
+              `set_up_maf` function.
+  #   exp_strategy: argument to specify whether to plot `wgs`, `wxs` samples or
+  #   `both`.
+  #   cosmic_clean_file: a file path to a TSV of COSMIC mutations that has been
+  #                     been cleaned up to have the genomic coordinates separated
+  #                     into Chr, Start, and End columns. This is passed to the
+  #                     `find_cosmic_overlap` function.
+  #
+  # Returns:
+  #  A violin plot with VAF plotted by whether or not it is overlapping with the
+  #  COSMIC mutation set.
+  #
+  # Make this argument case insensitive
+  exp_strategy <- toupper(exp_strategy)
+
+  # Filter out based on experimental stategy if specified
+  if (exp_strategy != "BOTH") {
+    vaf_df <- vaf_df %>%
+      dplyr::filter(experimental_strategy == exp_strategy)
+  }
+
+  # Get the overlap with COSMIC mutations using special function
+  maf_cosmic <- find_cosmic_overlap(vaf_df, cosmic_clean_file)
+
+  # Plot this as a violin plot
+  barplot <- maf_cosmic %>%
+    ggplot2::ggplot(ggplot2::aes(x = overlap_cosmic, y = vaf))
+
+  # Get rid of legend if both data aren't being plotted
+  if (exp_strategy != "BOTH") {
+    barplot <- barplot +
+      ggplot2::geom_violin(fill = "navyblue") +
+      ggplot2::theme(legend.position = "none")
+  } else {
+    barplot <- barplot +
+      ggplot2::geom_violin(ggplot2::aes(fill = experimental_strategy))
+  }
+  # Add some aesthetics
+  barplot <- barplot +
+    ggplot2::theme_classic() +
+    ggplot2::xlab("Same mutation and base change found in COSMIC") +
```

Comment:
Looking at the docs for `find_cosmic_overlap` it appears you are plotting the overlap here, not the same exact base change (that would be `same_cosmic`, right? 

Do you want to consider making this plot with both the `same_cosmic` and `overlap_cosmic` on the same plot?

Reply:
I could make both. IDK. This may be a thing that will be easier to make a decision on when we have the real data and I see what those plots look like.

Reply:
Agreed. But check the axis label...